### PR TITLE
[HER-28] Setup data project

### PR DESCRIPTION
**Pull request by TBiannca**
[HER-28](https://nerds-interns.atlassian.net/browse/HER-28?atlOrigin=eyJpIjoiNGI2MmQ3OGQ2ZjFkNDAyNGI4Mzc5ZjYzMTBjYzk1MTYiLCJwIjoiaiJ9)

### CODE REVIEW

**Review by danidoff — COMMENTED** [review]
Should we change Description and Related of the PR ? - hint: use "##"

**Comment by danidoff** [comment]
Should we rebase this PR to have the latest changes? Also, you will use the settings file from master.

File: src/Data/DataContext.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+using Microsoft.EntityFrameworkCore;
+
+namespace Data
+{
+    public class DataContext : DbContext
```
Comment (RaduBuciuceanu) [inline src/Data/DataContext.cs:5]: Would the name `Context` be more relevant here?

File: src/Presentation/Presentation.csproj [hunk 2]
```diff
@@ -4,5 +4,20 @@
         <TargetFramework>net5.0</TargetFramework>
         <Version>2021.0.0</Version>
     </PropertyGroup>
-    
+      <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore" Version="1.1.2" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="1.1.3" />
+    <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="1.1.2" />
```
Comment (RaduBuciuceanu) [inline src/Presentation/Presentation.csproj:10]: Do we really need all of those references here?

File: src/Presentation/Presentation.csproj [hunk 3]
```diff
@@ -4,5 +4,20 @@
...
+
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="1.1.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer.Design" Version="1.1.0" PrivateAssets="All" />
+    <PackageReference Include="MySql.EntityFrameworkCore" Version="5.0.3.1" />
```
Comment (RaduBuciuceanu) [inline src/Presentation/Presentation.csproj:14]: Shouldn't all of those refereces be placed inside `Data` project? @danidoff Can you help us here, please?

File: src/Presentation/Presentation.csproj [hunk 4]
```diff
@@ -4,5 +4,20 @@
...
+  </ItemGroup>
+  <ItemGroup>
+    <DotNetCliToolReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Tools" Version="1.0.2" />
+    <DotNetCliToolReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet" Version="1.0.0" />
+  </ItemGroup>
```
Comment (RaduBuciuceanu) [inline src/Presentation/Presentation.csproj:19]: Same here. Do we really need them? If yes, should we move them inside `Data`?

File: src/Presentation/Startup.cs [hunk 5]
```diff
@@ -1,21 +1,36 @@
...
 using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Hosting;
 
 namespace Presentation
 {
+    
     public class Startup
     {
+        
+        public string ConnectionString { get; set; }
```
Comment (RaduBuciuceanu) [inline src/Presentation/Startup.cs:16]: Why is it public? Do we use it outside of `Startup`?

Comment (RaduBuciuceanu) [reply src/Presentation/Startup.cs:16]: Should we also consider removing this field? Wouldn't be easier to use a local variable instead?

File: src/Presentation/Startup.cs [hunk 6]
```diff
@@ -1,21 +1,36 @@
...
         private readonly IConfiguration _config;
         
-        public Startup(IConfiguration config) =>_config = config;
+        public Startup(IConfiguration configuration)
+        {
+            _config = configuration;
+            Configuration = configuration;
+            ConnectionString = Configuration.GetConnectionString("DefaultConnection");
```
Comment (RaduBuciuceanu) [inline src/Presentation/Startup.cs:23]: Same here. Should we try to get rid of the `ConnectionString` property?

File: src/Presentation/Startup.cs [hunk 7]
```diff
@@ -1,21 +1,36 @@
...
+        }
         
-        public static void ConfigureServices(IServiceCollection services)
+       
+        public IConfiguration Configuration { get; }
```
Comment (RaduBuciuceanu) [inline src/Presentation/Startup.cs:27]: - Duplication of the already existent `_config` field. 
- It's public but it's used only inside `Startup`.

Should we remove it and use the `_config` field instead?

File: src/Presentation/Startup.cs [hunk 8]
```diff
@@ -1,21 +1,36 @@
...
+
+        public void ConfigureServices(IServiceCollection services)
         {
+            ConnectionString = Configuration.GetConnectionString("DefaultConnection");
```
Comment (RaduBuciuceanu) [inline src/Presentation/Startup.cs:31]: Should we try to fetch the value from inside `Settings/settings.default.json` instead? @danidoff :pray:

File: src/Presentation/appsettings.json [hunk 9]
```diff
@@ -6,5 +6,8 @@
       "Microsoft.Hosting.Lifetime": "Information"
     }
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "ConnectionStrings" : {
+    "DefaultConnectionString" : "server=localhost;userid=myusername;password=mypassword;database=mydatabase"
+  }
```
Comment (RaduBuciuceanu) [inline src/Presentation/appsettings.json:12]: Should we remove this whole file? Should we use the `Settings/settings.default.json` one instead?

**Review by RaduBuciuceanu — CHANGES_REQUESTED** [review]

**Review by RaduBuciuceanu — APPROVED** [review]

**Pull request merged**